aws-rds-optiongroup/aws-rds-optiongroup.json

```diff
@@ -0,0 +1,204 @@
+{
+  "typeName": "AWS::RDS::OptionGroup",
+  "description": "The AWS::RDS::OptionGroup resource creates an option group, to enable and configure features that are specific to a particular DB engine.",
+  "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds",
+  "definitions": {
+    "OptionConfiguration": {
+      "description": "The OptionConfiguration property type specifies an individual option, and its settings, within an AWS::RDS::OptionGroup resource.",
+      "type": "object",
+      "properties": {
+        "DBSecurityGroupMemberships": {
+          "description": "A list of DBSecurityGroupMembership name strings used for this option.",
+          "type": "array",
+          "uniqueItems": true,
+          "insertionOrder": false,
+          "items": {
+            "type": "string"
+          }
+        },
+        "OptionName": {
+          "description": "The configuration of options to include in a group.",
+          "type": "string",
+          "pattern": "^[a-zA-Z]{1}(?:-?[a-zA-Z0-9]){0,254}$"
```

Comment:
dont see regex in public docs - https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_OptionConfiguration.html. is it backward compatible?

Reply:
Removed from the spec. It was taken from a non-public source. I believe we can introduce this constraint later on but make it aligned with the public documentation.